src/servers/src/elasticsearch.rs

```diff
@@ -0,0 +1,362 @@
+
+// Return fake response for Elasticsearch ping request.
+#[axum_macros::debug_handler]
+pub async fn handle_get_version() -> impl IntoResponse {
+    let body = serde_json::json!({
+        "version": {
+            "number": ELASTICSEARCH_VERSION
+        }
+    });
+    (StatusCode::OK, elasticsearch_headers(), axum::Json(body))
+}
+
+// Return fake response for Elasticsearch license request.
+// Reference: https://www.elastic.co/guide/en/elasticsearch/reference/current/get-license.html.
+#[axum_macros::debug_handler]
+pub async fn handle_get_license() -> impl IntoResponse {
+    let body = serde_json::json!({
+        "license": {
+            "uid": "cbff45e7-c553-41f7-ae4f-9205eabd80xx",
+            "type": "oss",
+            "status": "active",
+            "expiry_date_in_millis": 4891198687000_i64,
+        }
+    });
+    (StatusCode::OK, elasticsearch_headers(), axum::Json(body))
+}
+
+// Process `_bulk` API requests. Only support to create logs.
+// Reference: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#docs-bulk-api-request.
+#[axum_macros::debug_handler]
+pub async fn handle_bulk_api(
+    State(log_state): State<LogState>,
+    Query(params): Query<LogIngesterQueryParams>,
+    Extension(mut query_ctx): Extension<QueryContext>,
+    TypedHeader(_content_type): TypedHeader<ContentType>,
+    payload: String,
+) -> impl IntoResponse {
+    let start = Instant::now();
+    debug!(
+        "Received bulk request, params: {:?}, payload: {:?}",
+        params, payload
+    );
+
+    let table = if let Some(table) = params.table {
+        table
+    } else {
+        return (
+            StatusCode::BAD_REQUEST,
+            elasticsearch_headers(),
+            axum::Json(write_bulk_response(
+                start.elapsed().as_millis() as i64,
+                0,
+                StatusCode::BAD_REQUEST,
+                "require parameter 'table'",
+            )),
+        );
+    };
+
+    // If pipeline_name is not provided, use the internal pipeline.
+    let pipeline = if let Some(pipeline) = params.pipeline_name {
+        pipeline
+    } else {
+        GREPTIME_INTERNAL_IDENTITY_PIPELINE_NAME.to_string()
+    };
+
+    // Read the ndjson payload and convert it to a vector of Value.
+    let log_values = match covert_es_input_to_log_values(&payload, &params.msg_field) {
+        Ok(log_values) => log_values,
+        Err(e) => {
+            return (
+                StatusCode::BAD_REQUEST,
+                elasticsearch_headers(),
+                axum::Json(write_bulk_response(
+                    start.elapsed().as_millis() as i64,
+                    0,
+                    StatusCode::BAD_REQUEST,
+                    e.to_string().as_str(),
+                )),
+            );
+        }
+    };
+    let log_num = log_values.len();
+
+    // The `schema` is already set in the query_ctx in auth process.
+    query_ctx.set_channel(Channel::Http);
```

Comment:
We can add a new channel type `elastic`. This http is mostly for our own protocols via http.